gpkitmodels/Commercial Sizing/CFP_Fuselage_Performance.py

```diff
@@ -0,0 +1,978 @@
+
+            #compute the mach number
+            V == M * a,
+            ])
+
+        #build the model
+        Model.__init__(self, None, constraints + self.atm, **kwargs)
+
+class Atmosphere(Model):
+    def __init__(self, **kwargs):
+        g = Variable('g', 'm/s^2', 'Gravitational acceleration')
+        p_sl = Variable("p_{sl}", "Pa", "Pressure at sea level")
+        T_sl = Variable("T_{sl}", "K", "Temperature at sea level")
+        L_atm = Variable("L_{atm}", "K/m", "Temperature lapse rate")
+        M_atm = Variable("M_{atm}", "kg/mol",
+                         "Molar mass of dry air")
+        p_atm = Variable("P_{atm}", "Pa", "air pressure")
+        R_atm = Variable("R_{atm}", "J/mol/K", "air specific heating value")
+        TH = 5.257386998354459 #(g*M_atm/R_atm/L_atm).value
+        rho = Variable('\\rho', 'kg/m^3', 'Density of air')
+        T_atm = Variable("T_{atm}", "K", "air temperature")
+        h = Variable("h", "m", "Altitude")
+
+        """
+        Dynamic viscosity (mu) as a function of temperature
+        References:
+        http://www-mdp.eng.cam.ac.uk/web/library/enginfo/aerothermal_dvd_only/aero/
+            atmos/atmos.html
+        http://www.cfd-online.com/Wiki/Sutherland's_law
+        """
+        mu  = Variable('\\mu',1.46*10**-5, 'kg/(m*s)', 'Dynamic viscosity')
+
+        T_s = Variable('T_s', 110.4, "K", "Sutherland Temperature")
+        C_1 = Variable('C_1', 1.458E-6, "kg/(m*s*K^0.5)",
+                       'Sutherland coefficient')
+        
+##        t_plus_ts_approx = (T_atm + T_s).mono_approximation({T_atm: 288.15,
+##                                                         T_s: T_s.value})
+
+        with SignomialsEnabled():
+            constraints = [
+                mu == mu,
+                # Pressure-altitude relation
+                (p_atm/p_sl)**(1/5.257) == T_atm/T_sl,
+
+                # Ideal gas law
+                rho == p_atm/(R_atm/M_atm*T_atm),
+
+                #temperature equation
+##                SignomialEquality(T_sl, T_atm + L_atm*h),
+                T_atm == 218*units('K'),
+
+                #constraint on mu
+##                SignomialEquality((T_atm + T_s) * mu, C_1 * T_atm**1.5),
+##                TCS([(T_atm + T_s) * mu >= C_1 * T_atm**1.5])
+                T_sl == 288.15*units('K'),
+                p_sl == 101325*units('Pa'),
+                R_atm == 8.31447*units('J/mol/K'),
+                M_atm == .0289644*units('kg/mol')
+                ]
+
+        #like to use a local subs here in the future
+        subs = None
+
+        Model.__init__(self, None, constraints, subs)
+
+class Engine(Model):
+    """
+    place holder engine model
+    """
+    def __init__(self, **kwargs):
+        #new variables
+        W_engine = Variable('W_{engine}', 'N', 'Weight of a Single Turbofan Engine')
+        
+        constraints = []
+
+        constraints.extend([
+            W_engine == 1000 * units('N')
```

Comment:
I think it'd be better practice to give the engine the default value of 1000 N, and have this constraint just be `W_engine == W_engine`